server/server.go

```diff
@@ -115,16 +117,30 @@ func (s *Server) newConn(conn net.Conn) *clientConn {
 		collation:    mysql.DefaultCollationID,
 		alloc:        arena.NewAllocator(32 * 1024),
 	}
-	log.Infof("[%d] new connection %s", cc.connectionID, conn.RemoteAddr().String())
+
 	if s.cfg.TCPKeepAlive {
 		if tcpConn, ok := conn.(*net.TCPConn); ok {
 			if err := tcpConn.SetKeepAlive(true); err != nil {
 				log.Error("failed to set tcp keep alive option:", err)
 			}
 		}
 	}
+
+	if s.proxyProtocolDecoder != nil {
+		// Proxy is configured. Read PROXY header first.
+		addr, err := s.proxyProtocolDecoder.readClientAddrBehindProxy(conn)
```

Comment:
yes. and we should NOT guessing the protocol according to spec for security considerations.